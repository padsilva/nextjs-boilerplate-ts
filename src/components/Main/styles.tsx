import styled from 'styled-components'

export const Wrapper = styled.main`
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
`

export const Background = styled.div`
  position: fixed;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  z-index: -1;
`

export const Title = styled.h1`
  font-size: 4rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 200;
`
