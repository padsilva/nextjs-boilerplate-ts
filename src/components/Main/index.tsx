import Image from 'next/image'

import * as S from './styles'

const Main = ({
  title = 'Boilerplate',
  description = 'TypeScript, ReactJS, NextJS and Styled Components'
}) => (
  <>
    <S.Background>
      <Image
        alt="Anonymous developer"
        src="/coding.jpg"
        layout="fill"
        objectFit="cover"
        quality={100}
      />
    </S.Background>
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
    </S.Wrapper>
  </>
)

export default Main
