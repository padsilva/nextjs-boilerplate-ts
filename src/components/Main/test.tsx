import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /boilerplate/i })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the text color correctly', () => {
    render(<Main />)

    expect(screen.getByRole('heading', { name: /boilerplate/i })).toHaveStyle({
      fontSize: '4rem'
    })
  })
})
