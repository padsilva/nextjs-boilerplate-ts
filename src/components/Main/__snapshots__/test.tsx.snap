// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c1 {
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  position: fixed;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
  z-index: -1;
}

.c2 {
  font-size: 4rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 200;
}

<div>
  <div
    class="c0"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt="Anonymous developer"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
    </div>
  </div>
  <main
    class="c1"
  >
    <h1
      class="c2"
    >
      Boilerplate
    </h1>
    <h2
      class="c3"
    >
      TypeScript, ReactJS, NextJS and Styled Components
    </h2>
  </main>
</div>
`
